#    Twitter Sentiment Analysis of the Bollywood Movie "Kabir Singh"

------
#### Author: [Monika Mishra](https://www.linkedin.com/in/monika-mishra-8b2a4115/)

#### Date: 08/14/2019

##### While browsing datasets on Kaggle, I stumbled upon the "Kabir Singh" dataset. It's a 2019 bollywood movie which focuses on the title character, an alcoholic surgeon who goes on a self-destructive path after his girlfriend is forced to marry someone else. The film was theatrically released in India on 21 June 2019 and received mixed reviews, with criticism directed at it for glamorising misogyny and toxic masculinity. But it did good in terms of revenue.

##### As I saw mixed reviews of the movie from my Facebook friends, I thought of analysing the twitter sentiment of the movie "Kabir Singh". Also this work combines my love for bollywood and data analytics. 😊 😎

#### Dataset Link : https://www.kaggle.com/shiri5h/tweets-bollywood-movie-kabir-singh

#### The tweet data is for the date range 21st June 2019 - 28th June 2019. The file size is 3.2 MB.

#### Dictionary dataset link : https://s3.amazonaws.com/hipicdatasets/dictionary.tsv

### First, import the libraries:

In [6]:
from pyspark.sql.types import *
from pyspark.sql.functions import *

### Load Source Data
Now load the data into a DataFrame. This data consists of the tweets related to "Kabir Singh" movie.
After the csv file is added to the data of the left frame, create a table using the UI.

In [8]:
data = sqlContext.sql("SELECT * FROM kabir")

data.show(5)

+-------------+------------------+--------------------+--------------------+--------------------+-----------+--------------+-------------+
 author| author_id| text| hashtags| mentions|reply_count|favorite_count|retweet_count|
+-------------+------------------+--------------------+--------------------+--------------------+-----------+--------------+-------------+
SharmaRavitha| 174397248| shahidkapoor I a...| []|['@shahidkapoor' ...| 0| 0| 0|
angry_rantman|940000000000000000|That's because yo...| []| []| 2| 14| 0|
pranaytikyani| 64400211|I’m obsessed with...| ['#KabirSingh']| []| 0| 2| 0|
 ishaqmb| 102953841|The character of ...|['#KabirSingh' '...|['@mentalhealth' ...| 0| 2| 1|
 BipinShukla1| 379698815|Advani_Kiara You ...| ['#KabirSingh']| ['@Advani_Kiara']| 0| 0| 0|
+-------------+------------------+--------------------+--------------------+--------------------+-----------+--------------+-------------+
only showing top 5 rows

In [9]:
print ("Number of records:", data.count())

Number of records: 12596

###Total number of tweets is 12,596

###Now let us see top 5 comments which have received the most likes.

In [11]:
%sql
SELECT author, text, favorite_count
FROM kabir
ORDER BY favorite_count desc limit 5

author text favorite_count taran_adarsh Update: KabirSingh is rocking... Will be Shahid Kapoor's biggest opener... Check out my video update: pic.twitter.com/QEVjTEhrMY 7822 taran_adarsh Uri. TotalDhamaal. LukaChuppi. Badla. AvengersEndgame. DeDePyaarDe. Now KabirSingh... Non-holiday releases yet successful... Proves yet again deliver qualitative content and non-holidays will turn into a celebration/festival for investors. Note: 2019 releases. 5731 ArmaanMalik22 Long time since shahidkapoor got his ACTUAL due. He’s literally one of the best actors of our industry. KabirSingh just reinstates that fact. Shash bhai you made me cry in places and pierced my heart in a few! Respect 5714 sonakshisinha Congratulations shahidkapoor Advani_Kiara and team KabirSingh for the amazing performance - on screen and box office!!!! 5101 FrustIndian Padmavat is just a movie not distortion of history.Haider is just a movie they are not insulting Indian army in real.PK is just a movie they are not insulting Hindu gods for real.But "Kabir Singh" movie is an Unapologetic Celebration of Toxic Masculinity. 4939

###That seems to be positive. 👍

### Let us check out the tweets which have been replied most.

In [13]:
%sql
SELECT author, text, reply_count
FROM kabir
ORDER BY reply_count desc limit 5

author text reply_count sonamohapatra & how can we keep such deeply disturbing dark & dangerous politics ‘aside’? Does the actor have no responsibility for choosing to play a part in a narrative that can set us back as a society? Is that all we have become? Creatures of ambition? LetsTalk India KabirSingh  … 473 taran_adarsh Update: KabirSingh is rocking... Will be Shahid Kapoor's biggest opener... Check out my video update: pic.twitter.com/QEVjTEhrMY 175 mymalishka KabirSingh is an ***hole .Women In The theatre whispering about what an ace S**T he is and He IS!Shahid is gold playing this flawed fella. Advani_Kiara big props.Loved ur work.I just hope nobody goes home thinking they want to BE Kabir or that they WANT a Kabir in their life. 168 kamaalrkhan Now question is this that if KabirSingh was released on Eid Toh Fir opening Kaya Hoti Aur Bharat Ka kaya Hotaa? 168 kamaalrkhan Beautiful innocent hot and sexy Advani_Kiara needed only one blockbuster to become the no.1 heroine of Bollywood and she has got Kabirsingh! So now nobody can stop her to finish career of Alia Bhatt and become the queen of Bollywood like Sridevi. 148

### OK.... that was kind of mixed reactions. 😐

### Let us check out the tweets which have been re-tweeted the most.

In [15]:
%sql
SELECT author, text, retweet_count
FROM kabir
ORDER BY retweet_count desc limit 5

author text retweet_count FrustIndian Padmavat is just a movie not distortion of history.Haider is just a movie they are not insulting Indian army in real.PK is just a movie they are not insulting Hindu gods for real.But "Kabir Singh" movie is an Unapologetic Celebration of Toxic Masculinity. 1892 taran_adarsh Update: KabirSingh is rocking... Will be Shahid Kapoor's biggest opener... Check out my video update: pic.twitter.com/QEVjTEhrMY 742 ArmaanMalik22 Long time since shahidkapoor got his ACTUAL due. He’s literally one of the best actors of our industry. KabirSingh just reinstates that fact. Shash bhai you made me cry in places and pierced my heart in a few! Respect 733 HaramiParindey People dissing Kabir Singh because it glorifies toxic masculinity. There’s a reason it is reel people are mature enough to know what not to follow. Those who aren’t a movie wont change their mindsetGangs of Wasseypur dekh ke aaye the to sabko maarne ka mann kar gaya tha kya?! 561 taran_adarsh Uri. TotalDhamaal. LukaChuppi. Badla. AvengersEndgame. DeDePyaarDe. Now KabirSingh... Non-holiday releases yet successful... Proves yet again deliver qualitative content and non-holidays will turn into a celebration/festival for investors. Note: 2019 releases. 542

### Again positive tweets 👏 👏 

### Now we will check the sentiments of the people through tweet text. 
#### *We will take the help of dictionary to check the polarity of words used in the tweet.

In [17]:
dictionary = sqlContext.sql("SELECT * FROM dictionary")

dictionary.show(5)

+----------+---+-----------+------+---------+---------+
 sub| a| word| type|attribute|sentiment|
+----------+---+-----------+------+---------+---------+
 weaksubj| 1| abandoned| adj| n| negative|
 weaksubj| 1|abandonment| noun| n| negative|
 weaksubj| 1| abandon| verb| y| negative|
strongsubj| 1| abase| verb| y| negative|
strongsubj| 1| abasement|anypos| y| negative|
+----------+---+-----------+------+---------+---------+
only showing top 5 rows

### Different views are created to calculate the polarity in a series of steps

In [19]:
%sql
create or replace view  v1 as
select author, texts
from kabir
lateral view explode(sentences(lower(text))) dummy as texts;

select * from v1 limit 5;

author texts SharmaRavitha List(shahidkapoor, i, am, in, love, with, kabir, kabirsingh, is, superb, movie, where, story, each, character, scene, aggression, flirting, love, made, it, worth, repeated, viewings) angry_rantman List(that's, because, you, are, sweet, teddy, bear, guy, and, chicks, magnet, and, if, you, behave, like, kabir, singh, all, girls, will, run, away) pranaytikyani List(i, m, obsessed, with, bekhayali, and, kaise, hua, kabirsingh) ishaqmb List(the, character, of, kabir, needs, a, psychological, treatment, instead, it's, glorified) ishaqmb List(wrong, message, to, audience, who, are, clapping, as, if, kabirs, mental, state, is, fun, mentalhealth, shahidkapoor, imvangasandeep, kamaalrkhan, kabirsingh, kabirsinghboxoffice, kabirsinghreview, mentalhealth)

In [20]:
%sql
create or replace view  v2 as
select author, word
from v1
lateral view explode(texts) dummy as word;

select * from v2 limit 5;

author word SharmaRavitha shahidkapoor SharmaRavitha i SharmaRavitha am SharmaRavitha in SharmaRavitha love

In [21]:
%sql
create or replace view v3 as 
select author, v2.word,
case d.sentiment
     when 'negative' then -1
     when 'positive' then 1
     else 0 
end as polarity
from v2 
left outer join dictionary d on v2.word = d.word;

select * from v3 limit 5;

author word polarity SharmaRavitha shahidkapoor 0 SharmaRavitha i 0 SharmaRavitha am 0 SharmaRavitha in 0 SharmaRavitha love 1

In [22]:
%sql
create or replace view v4 as
select author, 
case
when sum( polarity ) > 0 then 'positive'
when sum( polarity ) < 0 then 'negative'
else 'neutral' end as sentiment
from v3 group by author;

select * from v4 limit 5;

author sentiment BipinShukla1 positive RAJPARA_HARDIk_ positive MariaSartaj negative bellarkestark positive Iamnotmeiamwe positive

## Pie Chart for the sentiment

In [24]:
%sql
select sentiment, count(author) from v4 group by sentiment

sentiment count(author) positive 5388 neutral 2058 negative 1239

### Based on this dataset, 62 % people wrote positive tweets about the movie in a span of 7 days !! ✌️

## References

https://en.wikipedia.org/wiki/Kabir_Singh

https://www.kaggle.com/shiri5h/tweets-bollywood-movie-kabir-singh

## Acknowledgements

TweetScrape library in Python: https://github.com/5hirish/tweet_scrapper